urllib3/contrib/appengine.py

```diff
@@ -174,17 +179,46 @@ def urlopen(self, method, url, body=None, headers=None,
         http_response = self._urlfetch_response_to_http_response(
             response, retries=retries, **response_kw)
 
-        # Check for redirect response
-        if (http_response.get_redirect_location() and
-                retries.raise_on_redirect and redirect):
-            raise MaxRetryError(self, url, "too many redirects")
+        # Handle redirect?
+        redirect_location = redirect and http_response.get_redirect_location()
+        if redirect_location:
+            # Check for redirect response
+            if (self.urlfetch_retries and retries.raise_on_redirect):
+                raise MaxRetryError(self, url, "too many redirects")
+            else:
+                if http_response.status == 303:
+                    method = 'GET'
+
+                try:
+                    retries = retries.increment(method, url, response=http_response, _pool=self)
+                except MaxRetryError:
+                    if retries.raise_on_redirect:
+                        # Release the connection for this response, since we're not
+                        # returning it to be released manually.
+                        http_response.release_conn()
+                        raise
+                    return http_response
+
+                retries.sleep_for_retry(http_response)
+                log.info("Redirecting %s -> %s", url, redirect_location)
+                r = urlparse(redirect_location)
+                if not r.netloc:
+                    u = urlparse(url)
+                    redirect_url = Url(scheme=u.scheme, host=u.netloc, path=r.path).url
+                else:
+                    redirect_url = redirect_location
+                return self.urlopen(
+                    method, redirect_url, body, headers,
+                    retries=retries, redirect=redirect,
+                    timeout=timeout, **response_kw)
```

Comment:
@jonparrott `TestGAERetry` tests the case of `urlfetch_retries=True`. Could you please give me an example or explanation of the tests you want to see if you think that is not enough?